compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -395,16 +397,30 @@ BType resolvePatternTypeFromMatchExpr(BLangExpression matchExpr, BTupleType list
         return symTable.noType;
     }
 
-    BType resolvePatternTypeFromMatchExpr(BLangExpression matchExpr, BLangExpression constPatternExpr) {
+    public BType resolvePatternTypeFromMatchExpr(BLangErrorMatchPattern errorMatchPattern, BLangExpression matchExpr) {
         if (matchExpr == null) {
+            return errorMatchPattern.type;
+        }
+
+        if (isAssignable(matchExpr.type, errorMatchPattern.type)) {
```

Comment:
Shall we extract `matchExpr.type` and `errorMatchPattern.type` out to variables?

Reply:
Fixed